org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/refactoring/rename/RenameSupport.java

```diff
@@ -326,8 +327,10 @@ public static RenameSupport create(IField field, String newName, int flags) thro
 			return new RenameSupport(new RenameEnumConstProcessor(field), newName, flags);
 		} else {
 			final RenameFieldProcessor processor = new RenameFieldProcessor(field);
-			processor.setRenameGetter(updateGetterMethod(flags));
-			processor.setRenameSetter(updateSetterMethod(flags));
+			// https://github.com/redhat-developer/vscode-java/issues/2805
+			// 2. Refactor the field name should auto refactor all corresponding getter and setter of that field.
+			processor.setRenameGetter(JDTUtils.isGenerated(processor.getGetter()) ? true : updateGetterMethod(flags));
+			processor.setRenameSetter(JDTUtils.isGenerated(processor.getSetter()) ? true : updateGetterMethod(flags));
```

Comment:
Would it make more sense to change the flags argument at https://github.com/eclipse/eclipse.jdt.ls/blob/5cd372798427d35efd402efe3a18aad422702671/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/RenameHandler.java#L83 to `UPDATE_REFERENCES | UPDATE_GETTER_METHOD | UPDATE_SETTER_METHOD` ?

This would handle the case of lombok, but also any renaming of a field for which a getter/setter is detected. @fbricon , thoughts ?

Reply:
I think it might be a good idea

Reply:
Fixed.

Reply:
Did something change here ? I still see it as the same. I was expecting that instead of changing the logic in `RenameSupport`, you could just append the 2 additional getter/setter flags to get the rename to happen regardless of lombok or not.

Reply:
I have fixed it. See https://github.com/eclipse/eclipse.jdt.ls/pull/2339/commits/28ff9785fdc2c7bb9e9e9633d91db43735de4bab#diff-d71df854f8aac5ad0fdac7fcc1d8632b992ea0fcaa8c62d01f0a893775f1349fR84